R/calc_match_person.R

```diff
@@ -341,17 +341,20 @@ calc_match_person <- function(df_one, id_one, forename_one, surname_one, dob_one
 
     # Re-determine non-matches
     non_matches <- non_matches %>%
-      dplyr::anti_join(y = final_matches %>% dplyr::select(ID_ONE))
+      dplyr::anti_join(y = final_matches %>% dplyr::select(ID_ONE), by = "ID_ONE")
 
     # Re-determine total matches df
     matches <- matches %>%
       # Add exact matches
       dplyr::union_all(final_matches)
   } else {
+    # at this point no cross-join  is required as either all matched or would be too many records
+    if(cross_join_size > 10){
+      # Message if the cross-join is not viable
+      print("Final cross-join not viable as would require produce dataset greater than 10m rows")
+      print(paste0("Cross-join size: ", cross_join_size, " million rows"))
+    }
```

Comment:
Made sure message only triggered when cross-join was being applied and not when all records have been matched so no cross-join required